core/src/main/java/org/mule/runtime/core/internal/exception/ErrorHandler.java

```diff
@@ -170,4 +173,12 @@ public void setRootContainerName(String rootContainerName) {
     }
   }
 
+  public void setStatistics(FlowConstructStatistics flowStatistics) {
```

Comment:
Are you certain that at this point you'll have the injected exception listeners as well? (e.g.: OnCriticalErrorHandler)